src/server/controllers/checkPasswordToken.ts

```diff
@@ -205,24 +206,38 @@ export const checkTokenInOkta = async (
 			},
 		);
 
+		// attempt to determine if the user is using a native app
+		// first checking the pageData, then the prefix, otherwise undefined
+		const isNativeApp = state.pageData.isNativeApp
+			? state.pageData.isNativeApp
+			: prefix
+				? prefix.startsWith('a')
+					? 'android'
+					: 'ios'
+				: undefined;
+
+		// attempt to determine the app name
+		// first checking the pageData, then the prefix, otherwise undefined
+		const appName = state.pageData.appName
+			? state.pageData.appName
+			: prefix
+				? getAppName(prefix)
+				: undefined;
```

Comment:
This could be...

```suggestion
		const appName = state.pageData.appName ||
		    prefix
	                ? getAppName(prefix)
			: undefined;
```

Reply:
Used IIFE instead to make it more readable and explicit.